### DISCUSSION TITLE
CM Android Flutter build fails

### DISCUSSION THREAD

**Discussion by ZakDaMack** [codemagic-ci-cd/codemagic-docs#2578] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

The builds are able to run locally, this issue has occurred since flutter 3.22.1

1. echo "flutter.sdk=$HOME/programs/flutter" > "$CM_BUILD_DIR/android/local.properties"
2. $ANDROID_SDK_ROOT/tools/bin/sdkmanager --install "build-tools;34.0.0"
3. flutter build appbundle --release --build-name=$CM_TAG --build-number=$PROJECT_BUILD_NUMBER

### Expected results

Build should succeed

### Actual results

`ERROR:/Users/builder/.gradle/caches/transforms-3/281d4165655ea893dd7147c4b25ba565/transformed/vectordrawable-animated-1.2.0-runtime.jar: R8: java.lang.NullPointerException: Cannot invoke "String.length()" because "<parameter1>" is null`

`A failure occurred while executing com.android.build.gradle.internal.tasks.R8Task$R8Runnable
Compilation failed to complete, origin: /Users/builder/.gradle/caches/transforms-3/281d4165655ea893dd7147c4b25ba565/transformed/vectordrawable-animated-1.2.0-runtime.jar:androidx/vectordrawable/graphics/drawable/Animatable2Compat$AnimationCallback$1.class`

### Build id (optional)

665f1889bf1021c520b50b61

**Comment by benmartin88** [comment]
We are having the same problem. No changes, flutter 3.22.1. Builds just stopped working for android.

iOS unaffected.

**Comment by cerealexx** [comment]
same here...

**Comment by burhanrashid52** [comment]
Same here as well....

**Comment by benjamincombes** [comment]
Same here... But actually, I don't use Codemagic, it's the same with my vanilla project, I think it's more of a Flutter problem?
I just created an issue, if you want to +1 it here: https://github.com/flutter/flutter/issues/149798

**Reply by burhanrashid52** [reply]
Yeah, I don't use Codemagic as well. I am facing this on GitHub Action. Although the final error is the same I've had different logs before it. Will post that on your issue.

**Comment by benmartin88** [comment]
We managed to resolve our issue through a lot of trial and error. 

It seemed the error in CM was begin triggered by a specific flutter package that had received a minor patch (x.y.1). It wasn't possible to infer this from the minimal logging out of CM but we got this information from support (a list of version changes). 

For us, this was intercom_flutter and specifically 9.0.6. We locked this to 9.0.4 (last known working version). The failed builds coincide with the release of this package as we had built successfully 7 days prior, and 6 days prior Intercom had released the patch.

We are yet to raise this on their gitHub, but will be.

So if you literally have the same error and are using Intercom, then look at the package version. If not, some other package must be triggering the same build error. 

Now whether this is an Intercom issue, or a CM issue, I do not know. It builds just fine locally, so to me this suggests something in CM has changed that causes a dependancy issue when pulling that package version in their pipeline.

**Reply by benjamincombes** [reply]
Yes indeed, I can confirm I use Intercom package, I'll have a check right now, thanks for the info...!

**Reply by burhanrashid52** [reply]
@benmartin88 Yup we are using the intercom package as well. I locked it to `9.0.4` version and worked for me. Thanks a lot I've been struggling with this for the past 4 days.

**Reply by benmartin88** [reply]
Glad we could assist in resolving. I am still unsure whether the issues lies with Intercom or CM in terms of root cause, but I will raise on Intercoms GitHub at some point in the hope of finding out.

**Reply by huycozy** [reply, marked as answer]
it could be this issue: https://github.com/v3rm0n/intercom_flutter/issues/436 and OP shared his solution at https://github.com/v3rm0n/intercom_flutter/issues/436#issuecomment-2146614948.

**Reply by benjamincombes** [reply]
I can confirm, finally it works, thanks a lot @benmartin88 ..!!
